src/contexts/APIRouterContext/Provider.tsx

```diff
@@ -135,6 +137,55 @@ const APIRouterProvider: React.FC = ({ children }) => {
     [_handleSendAsset, accountRs, enqueueSnackbar, handleFetchAccountIDFromRS, publicKey]
   );
 
+  const _handlePlaceOrder = useCallback(
+    async (tx: IOrderPlacement, secretPhrase: string) => {
+      tx.secretPhrase = secretPhrase;
+      let result;
+      if (tx.orderType === "bid") {
+        result = await placeOrder(tx);
+      } else if (tx.orderType === "ask") {
+        result = await placeOrder(tx);
+      }
+      console.log("send asset result:", result);
+
+      if (!result) {
+        enqueueSnackbar(messageText.transaction.failure, { variant: "error" });
+        return;
+      }
+
+      enqueueSnackbar(messageText.transaction.success, { variant: "success" });
+    },
+    [enqueueSnackbar]
+  );
+
+  const handlePlaceOrder = useCallback(
+    async (orderType: "bid" | "ask", assetID: number, quantityQNT: BigNumber, priceNQT: BigNumber): Promise<true | undefined> => {
+      // TODO: validate quantity and price at the input layer, or here or somewhere smort
+
+      if (publicKey === undefined || accountRs === undefined) {
+        console.error("no public key or accountRs defined, returning...");
```

Comment:
snackbar instead of error here, something is significantly wrong as the user shouldn't be logged in if there's no rs or key

Reply:
fixed here and in a couple other spots in the same file